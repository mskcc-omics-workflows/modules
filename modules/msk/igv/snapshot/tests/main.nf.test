nextflow_process {

    name "Test Process IGV_SNAPSHOT"
    script "../main.nf"
    process "IGV_SNAPSHOT"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "igv"
    tag "igv/snapshot"

    test("chr22 - bam, vcf") {

        when {
            process {
                """
                input[0] = [
                    [ id:'C-9M47TU-L001-d'], // meta map
                    file(params.test_data_mskcc['chr22']['chr22bam'], checkIfExists: true),
                    file(params.test_data_mskcc['chr22']['chr22bam_bai'], checkIfExists: true),
                    file(params.test_data_mskcc['chr22']['chr22vcf'], checkIfExists: true)

                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("chr22 - bam, vcf - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'C-9M47TU-L001-d'], // meta map
                    file(params.test_data_mskcc['chr22']['chr22bam'], checkIfExists: true),
                    file(params.test_data_mskcc['chr22']['chr22bam_bai'], checkIfExists: true),
                    file(params.test_data_mskcc['chr22']['chr22vcf'], checkIfExists: true)
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
