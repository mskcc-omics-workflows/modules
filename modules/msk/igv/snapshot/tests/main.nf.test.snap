{
    "chr22 - bam, vcf - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "C-9M47TU-L001-d"
                        },
                        [
                            "test.bat:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,870a90d9f64f5a5f5ec4cd474d69027c"
                ],
                "igv_screenshots": [
                    [
                        {
                            "id": "C-9M47TU-L001-d"
                        },
                        [
                            "test.bat:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test.png:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,870a90d9f64f5a5f5ec4cd474d69027c"
                ]
            }
        ],
        "timestamp": "2024-08-06T17:08:44.477177"
    },
    "chr22 - bam, vcf": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "C-9M47TU-L001-d"
                        },
                        [
                            "22_22127176_22127176.png:md5,fd9de31a371db4523f8a0982df749a09",
                            "22_41565827_41565827.png:md5,1904ecf0b28baeee6e14e78543052a17",
                            "22_41566406_41566406.png:md5,c77c48716ed2272f45b49c7852a74705",
                            "22_41566413_41566413.png:md5,bfaf4189292aa0495e51c26ea33f64f7",
                            "22_41566595_41566595.png:md5,2fca6b6e75fd4c533fb512d0cd11941b",
                            "22_41566657_41566657.png:md5,fd49c1c6439206da9c15595921392d29",
                            "IGV_snapshots.bat:md5,d2eaf7a013f62e5fe07f8a0ddf459ace"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,870a90d9f64f5a5f5ec4cd474d69027c"
                ],
                "igv_screenshots": [
                    [
                        {
                            "id": "C-9M47TU-L001-d"
                        },
                        [
                            "22_22127176_22127176.png:md5,fd9de31a371db4523f8a0982df749a09",
                            "22_41565827_41565827.png:md5,1904ecf0b28baeee6e14e78543052a17",
                            "22_41566406_41566406.png:md5,c77c48716ed2272f45b49c7852a74705",
                            "22_41566413_41566413.png:md5,bfaf4189292aa0495e51c26ea33f64f7",
                            "22_41566595_41566595.png:md5,2fca6b6e75fd4c533fb512d0cd11941b",
                            "22_41566657_41566657.png:md5,fd49c1c6439206da9c15595921392d29",
                            "IGV_snapshots.bat:md5,d2eaf7a013f62e5fe07f8a0ddf459ace"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,870a90d9f64f5a5f5ec4cd474d69027c"
                ]
            }
        ],
        "timestamp": "2024-08-06T17:08:31.32318"
    }
}